internal/server/server.go

```diff
@@ -92,6 +99,18 @@ func (s *Server) ListenAndServe(addr string) error {
 	return http.ListenAndServe(addr, s.router)
 }
 
+func (s *Server) WithConfig(config *config.Root) *Server {
```

Comment:
❓ How about this, what do you think? We'll only need the service bits.
```suggestion
func (s *Server) WithConfig(config *config.Service) *Server {
```